src/typescript/mit-open-api-axios/tsconfig.json

```diff
@@ -0,0 +1,109 @@
+{
+  "compilerOptions": {
+    /* Visit https://aka.ms/tsconfig to read more about this file */
+
+    /* Projects */
+    // "incremental": true,                              /* Save .tsbuildinfo files to allow for incremental compilation of projects. */
+    // "composite": true,                                /* Enable constraints that allow a TypeScript project to be used with project references. */
+    // "tsBuildInfoFile": "./.tsbuildinfo",              /* Specify the path to .tsbuildinfo incremental compilation file. */
+    // "disableSourceOfProjectReferenceRedirect": true,  /* Disable preferring source files instead of declaration files when referencing composite projects. */
+    // "disableSolutionSearching": true,                 /* Opt a project out of multi-project reference checking when editing. */
+    // "disableReferencedProjectLoad": true,             /* Reduce the number of projects loaded automatically by TypeScript. */
+
+    /* Language and Environment */
+    "target": "es6",                                     /* Set the JavaScript language version for emitted JavaScript and include compatible library declarations. */
+    // "lib": [],                                        /* Specify a set of bundled library declaration files that describe the target runtime environment. */
+    // "jsx": "preserve",                                /* Specify what JSX code is generated. */
+    // "experimentalDecorators": true,                   /* Enable experimental support for legacy experimental decorators. */
+    // "emitDecoratorMetadata": true,                    /* Emit design-type metadata for decorated declarations in source files. */
+    // "jsxFactory": "",                                 /* Specify the JSX factory function used when targeting React JSX emit, e.g. 'React.createElement' or 'h'. */
+    // "jsxFragmentFactory": "",                         /* Specify the JSX Fragment reference used for fragments when targeting React JSX emit e.g. 'React.Fragment' or 'Fragment'. */
+    // "jsxImportSource": "",                            /* Specify module specifier used to import the JSX factory functions when using 'jsx: react-jsx*'. */
+    // "reactNamespace": "",                             /* Specify the object invoked for 'createElement'. This only applies when targeting 'react' JSX emit. */
+    // "noLib": true,                                    /* Disable including any library files, including the default lib.d.ts. */
+    // "useDefineForClassFields": true,                  /* Emit ECMAScript-standard-compliant class fields. */
+    // "moduleDetection": "auto",                        /* Control what method is used to detect module-format JS files. */
+
+    /* Modules */
+    "module": "commonjs",                                /* Specify what module code is generated. */
```

Comment:
I think we should change this to 
```
"module": "es6",                     // use import/export syntax
"moduleResolution": "node10"         // traditional resolution
```
`"module": "es6"` results in import/export syntax in the transpiled code; `"moduleResolution": "node10"` is "traditional" module/file resolution, which doesn't require file extensions and allows "bare" imports from node_modules (like `"import "axios"`). We **need** to use `"moduleResolution": "node"10"` based on the generated code, but "module" field is our choice.

(I think the default value for moduleResolution depends on the value of module; if you change to `"es6"` without changing moduleResolution, I think TS will error.)

### Rationale:
I did a quick experiment with bundling and tree-shaking, changing `App.tsx` to
```ts
import { v1 } from "@mitodl/open-api-axios"
window.articlesApi = v1.ArticlesApi
```
so that only `@mitodl/open-api-axios` and its dependencies would be included in MIT Open's production bundle.

- with commonJS, **all** of mitodl/open-api-axios is included; ~155kb (w/ axios)
- with es6 module syntax, only `ArticlesApi` (and the common code) was included, ~35 kb  (w/ axios)

Tree-shaking is supposed to work in webpack5 with the old commonjs syntax...but it doesn't actually seem to be working. I'm not sure if that has to do with exactly how TS transpiles to cjs, or if it's because of the star imports, or what.

But... I think es6 syntax would be better, and quick experiment bears that out.